[Al-ez] iP

Al-ez opened the pull request:
# Meow 

> "You cannot confuse the enemy, if you are not confused yourself." - Confucius

**Summary**: Your all in one chatbot, that aims to confuse absolutely nobody. 

## Key Features
- Enhanced date and time parsing
- Improved task management
- Super Cool

### How to use
1. Download it [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ)
2. Add your tasks.
3. Use it like a PRO 😸 :

Features:

- [x] Add tasks
- [x] Add Deadlines
- [x] Add Events
- [ ] User Interface (coming soon)

---
## Introducing our cutting edge technology --- THE DATE-TIME-PARSER!!!! 🥇 
```java
// Example of enhanced date parsing
LocalDate date = LocalDate.parse("2023-09-04");

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Al-ez committed:
Completed increment Level-0

Al-ez committed:
Level-0

Al-ez committed:
Level-1

Al-ez committed:
Added list and add to list function

Al-ez committed:
Added list and add to list function

Al-ez committed:
Add Deadline, Event & ToDo class

Al-ez committed:
Add MeowException and handle exceptions

Al-ez committed:
Add delete function, change Task[] to ArrayList<Task>

Al-ez committed:
Do automated text UI testing

Al-ez committed:
Implement save function

Al-ez committed:
Merge branch-Level-7 into master

Al-ez committed:
Implement date function

Al-ez committed:
Merge branch-Level-8 into master

Al-ez committed:
Refactor Code for OOP

Al-ez committed:
Put all classes in duke package

Al-ez committed:
Merge remote-tracking branch 'origin/add-gradle-support'
Added Gradle build support to the project

- Integrated Gradle to automate build tasks
- Added Gradle configuration files (e.g., build.gradle)
- Updated project structure to support Gradle builds

Al-ez committed:
Add JUnit tests for Parser and TaskList

Al-ez committed:
Create Jar file

Al-ez committed:
Add Javadocs to all classes

Al-ez committed:
Reformat code to coding standard

Al-ez committed:
Add find function

Al-ez committed:
Add find function

jian7490 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,210 @@
+ * as well as viewing outstanding tasks.
+ */
+public class Parser {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd");
```
Good to see making a frequently used variable constant and it is following its naming convention.

jian7490 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,210 @@
+                        throw new MeowException("GRRR! Invalid task number, you only have " + taskList.getTaskCount()
+                                + (taskList.getTaskCount() == 1 ? " task." : " tasks."));
+                    }
+                } else { // "mark"
+                    throw new MeowException("Please specify which task to mark. Example: mark 1");
```
May I know why is the "mark" comment put here at the throw-exception clause?

jian7490 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,210 @@
+                        try {
+                            LocalDate by = parseDate(parts[1].trim());
+                            taskList.addTask(new Deadline(parts[0], by));
+                            ui.showMessage("Got it. I've added this task:\n"
+                                    + taskList.getTask(taskList.getTaskCount() - 1));
+                            storage.saveTasks(taskList.getTasks());
+                            ui.showMessage(taskList.getTaskCount() <= 1
+                                    ? "Now you have " + taskList.getTaskCount() + " task in the list."
+                                    : "Now you have " + taskList.getTaskCount() + " tasks in the list.");
```
It is nice to break long lines into shorter ones here!

jian7490 reviewed (commented):
Overall the code is clean and follows the coding standards. Check the import statements to make them consistent and avoid wildcard.

sandyk0105 commented on src/main/java/Deadline.java:
```diff
@@ -1,16 +1,19 @@
     @Override
     public String toString() {
-        return "[D]" + super.toString() + " (by: " + by + ")";
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy");
```
What about the time? How does the bot accept the time of the deadline / event?

Al-ez replied:
If im not wrong the requirements said use LocalDate, but LocalDate cannot store time values

Al-ez committed:
Fix peer feedback

Al-ez committed:
Add GUI

Al-ez committed:
Fix date parsing logic to handle invalid dates

- Added a check for invalid date formats like "2020-31-31" that cause DateTimeParseException.
- Changed the catch block to return a message to the user instead of printing to the console.
- Improved error message output to inform users of invalid input formats.

Add assertions in deleteTask method for index bounds

- Added assert to ensure the index is within valid range before deleting a task.
- This ensures that any invalid index values are caught during testing or development.
- Improves code robustness and prevents potential runtime errors.

Display welcome message on chatbot start

- Added logic in the initialize method to display a welcome message as soon as the chatbot is launched.
- This improves user experience by providing an initial response without needing user input.
- Updated UI layout to accommodate the welcome message properly.

Al-ez committed:
Fix date parsing logic to handle invalid dates

- Added a check for invalid date formats like "2020-31-31" that cause DateTimeParseException.
- Changed the catch block to return a message to the user instead of printing to the console.
- Improved error message output to inform users of invalid input formats.

Add assertions in deleteTask method for index bounds

- Added assert to ensure the index is within valid range before deleting a task.
- This ensures that any invalid index values are caught during testing or development.
- Improves code robustness and prevents potential runtime errors.

Display welcome message on chatbot start

- Added logic in the initialize method to display a welcome message as soon as the chatbot is launched.
- This improves user experience by providing an initial response without needing user input.
- Updated UI layout to accommodate the welcome message properly.

Al-ez committed:
Refactor Parser.java and Ui.java

- Simplified method implementations for clearer logic
- Removed redundant code and comments
- Changed return type of methods to fit GUI

These changes help make the codebase more readable.

Al-ez committed:
Merge pull request #1 from Al-ez/A-Assertions

A assertions

Al-ez committed:
Merge branch 'master' into A-CodeQuality

Al-ez committed:
Fix minor bugs

Al-ez committed:
Fix minor bugs

Al-ez committed:
Merge branch 'A-CodeQuality' into master

Al-ez committed:
Fix minor bugs

Al-ez committed:
Add extension C-Sort

Al-ez committed:
Add C-Sort and fix bugs

Al-ez committed:
Minor bug fixes

Al-ez committed:
Refactor long method by extracting helper methods

Abstracted a long method into smaller, more manageable helper methods.
This improves code readability and maintainability by encapsulating
distinct functionality into separate methods. Each helper method
handles a specific part of the original method's logic.

Al-ez committed:
Modify JavaDoc

Slight rephrasing of Javadoc in ToDo class, toString method

Al-ez committed:
Modify JavaDoc

Added @return at in JavaDoc at Ui class, getWelcomeMessage() method

Al-ez committed:
Modify JavaDoc

Add @return to getDescription() method in Task class

Al-ez committed:
Add JavaDoc

Add JavaDoc to toString() method in Task class

Al-ez committed:
Add code reuse

Cited source for GUI components. Code taken from CS2103T website.

Al-ez committed:
Create user guide

Update README.md and included instructions on product use

Al-ez committed:
Create user guide

Updated the readme and created a link to the release

Al-ez committed:
Update readme

Changed default duke readme to new meow readme

Al-ez committed:
FIx error in user guide
